apps/site/lib/site_web/controllers/route_controller.ex

```diff
@@ -0,0 +1,12 @@
+defmodule SiteWeb.RouteController do
+  @moduledoc """
+  Endpoints for getting route data.
+  """
+  use SiteWeb, :controller
+  alias Routes.Repo
+
+  def get_by_stop_id(conn, %{"stop_id" => stop_id} = _params) do
+    routes = Repo.by_stop(stop_id)
+    json(conn, routes)
```

Comment:
**idea:** the header doesn't need most of the route information; have you considered having this backend endpoint take these routes and compute and return a list of icon names directly?

Reply:
I was trying to find it.  I think the `meta` tags used by the stop redesign page should live on the `show-redesign.html.eex` instead of the `app.html.eex` page.
I don't like the approach of embedding all of the data into the page.  I think its more flexible to fetch the data on loading of the page.  The header icons need more than IDs.  We would either have to fetch the routes by id on load or embed all the of the routes into the page.

As for getting a list of icons from the backend.  I see it as a design decision.  Do we want to do most of the data processing in the backend or front end?  My only thought is the more front end processing the backend does the more tightly coupled they could become.

Reply:
> I think the meta tags used by the stop redesign page should live on the show-redesign.html.eex instead of the app.html.eex page.

I see your point there, and your point against embedding the data in the page. That's fine.

> As for getting a list of icons from the backend. I see it as a design decision. Do we want to do most of the data processing in the backend or front end? My only thought is the more front end processing the backend does the more tightly coupled they could become.

That's an interesting point about coupling; I hadn't thought about the backend as doing "front end processing" vs. (normal?) processing. I'm not sure that's a meaningful distinction in the case of Dotcom - where processing stuff for the whims of the website UI is kind of the whole name of the game! 

Where practical, I prefer more work to happen on the backend, simply because of its greater computing power, and also because it enables us to use much simpler front-end functions. There are many cases where we currently use the backend to do more of the heavy lifting - one recent example would be the hours of operation, which isn't an object that exists in the V3 API, but is crafted meets a specific use case for our frontend. 

Not to dismiss the potential dangers of being tightly coupled, but in reality Dotcom is a loooong way from being a loosely coupled backend + frontend setup.

Reply:
I looked into changing this to create an endpoint that just returned icons needed, and it felt like overkill.  We aren't processing the data so much as parsing it.  The stop object we already have from the backend has half of the information we need to parse out the icons.  The other information is in the route list, which I think the stop page is going to need anyway.

It felt like the wrong approach to make an endpoint just to get half the data (which lines connect to the stop), and recompute half the data we already have (Such as CR zone number, accessibility, and Parking).